spec/support/examples_group/shared_examples_for_aggregation_mixin.rb

```diff
@@ -0,0 +1,54 @@
+shared_examples_for "AggregationMixin" do |through|
+  context "includes AggregationMixin" do
+    include Spec::Support::ArelHelper
+
+    let(:hardware) do
+      FactoryBot.create(:hardware,
+                        :cpu2x2,
+                        :ram1GB,
+                        :cpu_speed     => 2_999,
+                        :disk_capacity => 40,
+                        :memory_mb     => 2048)
+    end
+
+    let(:ems) { FactoryBot.create(:ext_management_system) }
+    let!(:vm) { FactoryBot.create(:vm, :hardware => hardware, :ext_management_system => ems) }
+    let!(:vm1) { FactoryBot.create(:vm, :hardware => hardware, :ext_management_system => ems) }
+    let!(:host) { FactoryBot.create(:host, :hardware => hardware, :ext_management_system => ems) }
+    let!(:host1) { FactoryBot.create(:host, :hardware => hardware, :ext_management_system => ems) }
+    let(:object) { FactoryBot.create(described_class.to_s.underscore.to_sym) }
+
+    before do
+      if through == "ext_management_systems"
+        object.ext_management_systems << ems
+      elsif through == "computer_systems"
+        object.computer_system_hardwares << [hardware]
+      elsif described_class == Host
+        object.host_hardwares << [hardware]
+        object.vms_and_templates << [vm, vm1]
+      else
+        object.hosts << [host, host1]
+        object.vms_and_templates << [vm, vm1]
+      end
+    end
+
+    describe "calculates single object" do
+      context "host" do
+        it "calculates hardware specs" do
```

Comment:
These are great!  Thanks!

One critique though:  Can you make these individual `it` blocks instead of as one block:

```
it "calculates #aggregate_cpu_speed" do
  expect(object.aggregate_cpu_speed).to eq(11_996)
end

it "calculates #aggregate_cpu_total_cores" do
  expect(object.aggregate_cpu_total_cores).to eq(4)
end

# etc
```

http://www.betterspecs.org/#single

Reply:
Also, do you think it makes sense to make use of some of these tests as well?

https://github.com/ManageIQ/manageiq/blob/master/spec/support/custom_matchers/exceed_query_limit.rb
https://github.com/ManageIQ/manageiq/blob/master/spec/support/virtual_attribute_helper.rb [[ref](https://github.com/ManageIQ/activerecord-virtual_attributes/blob/master/lib/active_record/virtual_attributes/rspec/have_virtual_attribute.rb)]